### PR TITLE
feat(intersection): Add `intersection` to compat layer

### PR DESCRIPTION
**Pull request by mass2527**
I added the `intersection` function to the compatibility layer.

스크린샷 2024-10-13 오후 10 14 03

### CODE REVIEW

File: docs/ja/reference/array/intersection.md [hunk 1]
```diff
@@ -28,3 +28,25 @@ const array2 = [3, 4, 5, 6, 7];
 const result = intersection(array1, array2);
 // 両方の配列に含まれる [3, 4, 5] を返します。
 ```
+
+## Lodashとの互換性
+
+`es-toolkit/compat`から`intersection`をインポートすると、lodashと互換性があります。
+
+- `intersection`は公通要素を見つけるために複数の配列のようなオブジェクトを受け入れます。
```
Comment (raon0211) [inline docs/ja/reference/array/intersection.md:36]:
```suggestion
- `intersection`は公通要素を見つけるために複数の配列風オブジェクト(Array-like object)を受け入れます。
```

File: benchmarks/performance/intersection.bench.ts [hunk 2]
```diff
@@ -13,6 +15,10 @@ describe('intersection, small arrays', () => {
     intersectionToolkit(array1, array2);
   });
 
+  bench('es-toolkit/compat/difference', () => {
```
Comment (mass2527) [inline benchmarks/performance/intersection.bench.ts:18]:
```suggestion
  bench('es-toolkit/compat/intersection', () => {
```

File: benchmarks/performance/intersection.bench.ts [hunk 3]
```diff
@@ -26,6 +32,10 @@ describe('intersection, large arrays', () => {
...
```
Comment (mass2527) [inline benchmarks/performance/intersection.bench.ts:35]:
```suggestion
  bench('es-toolkit/compat/intersection', () => {
```

File: src/compat/array/intersection.spec.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const object = {};
+    const largeArray = times(LARGE_ARRAY_SIZE, constant(object));
+
+    expect(intersection([object], largeArray)).toEqual([object]);
+    expect(intersection(range(LARGE_ARRAY_SIZE), [1])).toEqual([1]);
+  });
+
+  it('should treat values that are not arrays or `arguments` objects as empty', () => {
+    const array = [0, 1, null, 3];
+    expect(intersection(array, 3, { 0: 1 }, null)).toEqual([]);
```
Comment (raon0211) [inline src/compat/array/intersection.spec.ts:84]:
```suggestion
    // @ts-ignore
    expect(intersection(array, 3, { 0: 1 }, null)).toEqual([]);
```

**Review by raon0211 — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**